qiskit/passmanager/passmanager.py

```diff
@@ -0,0 +1,264 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Manager for a set of Passes and their scheduling during transpilation."""
+
+from abc import ABC
+from typing import List, Callable, Any, Union, Optional, Sequence
+import logging
+import dill
+from qiskit.tools.parallel import parallel_map
+
+from .base_pass import BasePass
+from .passrunner import BasePassRunner
+from .exceptions import PassManagerError
+from .flow_controllers import FlowController, PassSequence
+
+logger = logging.getLogger(__name__)
+
+
+class BasePassManager(ABC):
```

Comment:
Actually a developer needs to attach `BasePassManager.PASS_RUNNER` to make it work. In this base class `BasePassRunner` is just added for convenience of type hint, but the base class doesn't provide IR conversion logic. One could also do abstract class property, but I don't like class property. 

Alternatively, we can keep `BasePassManager._create_running_passmanager` method and make it abstract. However, writing and overriding this method just for difference of pass runner type seems awkward (we can remove this method in the followup PR #10127 ). 

The another place where the pass runner type is necessary is [run](https://github.com/Qiskit/qiskit-terra/pull/10124/files#diff-9432898faf9979ca2ac7448bfd24071382f544b42aac863790fe965d1dd1f3d2R216-R218) method. We need to dispatch the `_run_single_circuit` when input is a single entry, but we cannot safely count entry number because input program is usually iterable, e.g. `QuantumCircuit` and `ScheduleBlock`.